src/dex/generic-rfq/generic-rfq.ts

```diff
@@ -323,126 +324,131 @@ export class GenericRFQ extends ParaSwapLimitOrders {
-          side,
-          requiredAmountWithSlippage,
-          makerAssetAmountFilled.toString(),
-          slippageFactor,
-        );
+        if (makerAssetAmountFilled < BigInt(requiredAmountWithSlippage)) {
+          throw new SlippageCheckError(
+            this.dexKey,
+            this.network,
+            side,
+            requiredAmountWithSlippage,
+            makerAssetAmountFilled.toString(),
+            slippageFactor,
+          );
+        }
+      } else {
+        if (makerAssetAmount < destAmount) {
+          throw new SlippageCheckError(
+            this.dexKey,
+            this.network,
+            side,
+            destAmount.toString(),
+            makerAssetAmount.toString(),
+            slippageFactor,
+            true,
+          );
+        }
+
+        const requiredAmountWithSlippage = new BigNumber(srcAmount.toString())
+          .multipliedBy(slippageFactor)
+          .toFixed(0);
+
+        if (takerAssetAmount > BigInt(requiredAmountWithSlippage)) {
+          throw new SlippageCheckError(
+            this.dexKey,
+            this.network,
+            side,
+            requiredAmountWithSlippage,
+            takerAssetAmount.toString(),
+            slippageFactor,
+          );
+        }
       }
-    } else {
-      if (makerAssetAmount < destAmount) {
-        throw new SlippageCheckError(
-          this.dexKey,
-          this.network,
-          side,
-          destAmount.toString(),
-          makerAssetAmount.toString(),
-          slippageFactor,
-          true,
+
+      let isApproved = false;
+
+      // isApproved is only used in direct method and available only for v6, then no need to check approve for v5
+      // because it's either done in getSimpleParam or approve call in the adapter smart contract
+      if (options.version === ParaSwapVersion.V6 && options.isDirectMethod) {
+        isApproved = await this.dexHelper.augustusApprovals.hasApproval(
+          options.executionContractAddress,
+          // ETH always need to be wrapped for RFQ
+          this.dexHelper.config.wrapETH(srcToken).address,
+          this.augustusRFQAddress,
         );
       }
 
-      const requiredAmountWithSlippage = new BigNumber(srcAmount.toString())
-        .multipliedBy(slippageFactor)
-        .toFixed(0);
-
-      if (takerAssetAmount > BigInt(requiredAmountWithSlippage)) {
-        throw new SlippageCheckError(
-          this.dexKey,
-          this.network,
-          side,
-          requiredAmountWithSlippage,
-          takerAssetAmount.toString(),
-          slippageFactor,
+      return [
+        { ...optimalSwapExchange, data: { orderInfos: [order], isApproved } },
+        { deadline: minDeadline },
+      ];
+    } catch (e) {
+      if (e instanceof SlippageCheckError) {
+        this.logger.warn(
+          `${this.dexKey}-${this.network}: failed to build transaction on side ${side} with too strict slippage. Skipping restriction`,
+        );
+      } else {
```

Comment:
let's blacklist passed user address if MM returned `403` status code, this update was partially communicated with MMs